### DISCUSSION TITLE
many-to-many relationships

### DISCUSSION THREAD

**Discussion by kelbyfaessler** [planetscale/discussion#411] [Q&A]
Planetscale doesn't allow for foreign key constraints at the database level (https://planetscale.com/docs/learn/operating-without-foreign-key-constraints).

So we have to implement foreign keys ourselves and manage the integrity at the application level. Ok, no big deal considering what planetscale gives us.

Does this also apply to many-to-many relationships? Since those would require an associative table that uses two FKs?

Prisma has implicit many-to-many relationships where you don't have to define the associative table in your prisma schema and prisma will manage it for you. Are we still able to use this prisma feature with planetscale? Or will planetscale require me to manually define my associative tables in the schema?

Maybe since we define `relationMode` in the prisma settings when working with planetscale, it'll work automatically?

**Comment by kblizeck** [comment]
I have the same question/concern!

I have multiple Prisma implicit many-to-many relationships, and I'm discovering that they do not delete as expected... 
```
model User {
  id                Int                       @id @default(autoincrement())
  firstName         String?
  lastName          String?
  email             String?                   @unique
  audiences         Audience[]
  createdAt         DateTime                  @default(now())
}

model Audience {
  id        Int        @id @default(autoincrement())
  name      String     @unique
  users     User[]
}
```

If I delete a User, the association is still present on the `Audience` table under `users` - though the `User` record doesn't exist. 

Does this mean I have to explicitly define all my many-to-many relationships, just so I can go around the foreign key constraints and define the `onDelete` methods for each?

**Comment by kblizeck** [comment]
> Maybe since we define `relationMode` in the prisma settings when working with planetscale, it'll work automatically?

You can't use `relationMode` in prisma in [implicit many-to-many relationships](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations), so I'm assuming that we _have_ to explicitly create our many-to-many's so we can define the `relationMode` expectations.
![Screen Shot 2023-03-15 at 5 17 41 PM](https://user-images.githubusercontent.com/9886123/225465127-b82aa35f-8b0f-4c42-9531-eaa775130c73.png)

**Reply by John-Dennehy** [reply]
Having the same confusion using drizzle, trying to create a suitable select query.
